blockchain/blockchain.go

```diff
@@ -177,6 +179,13 @@ func ClockOption(clk clock.Clock) Option {
 	}
 }
 
+func BlockTimeCalculatorBuilderOption(builder *blockutil.BlockTimeCalculatorBuilder) Option {
+	return func(bc *blockchain) error {
+		bc.btcBuilder = builder
+		return nil
+	}
+}
```

Comment:
what's actually passed in by this option is the `getBlockInterval` func, the other 2 funcs `getTipHeight` and `getHistoryBlockTime`, so it's more proper to call it `BlockIntervalTimeOption`, like below? and can directly build the time calculator here:
```
func BlockIntervalTimeOption(getBlockInterval func(uint64) time.Duration) Option {
    return func(bc *blockchain) error {
        bc.btcBuilder, err = NewBlockTimeCalculator(getBlockInterval, bc.TipHeight, ...)
        return err
    }
}
```